File: cpp/0665-Non-Decreasing-Array.cpp
class Solution {
public:
    bool checkPossibility(vector<int>& nums) {
        int count = 0;
        for(int i=1;i<nums.size();i++){
            if(nums[i-1]>nums[i]){
                count++;
                if(i>=2&&nums[i-2]>nums[i]){
                    nums[i]=nums[i-1];
                }
                else{
                    nums[i-1]=nums[i];
                }
            }
        }
        return count<=1;
    }
};

File: kotlin/0665-non-decreasing-array.kt
class Solution {
    fun checkPossibility(nums: IntArray): Boolean {
        var modified = false
        for(i in 0 until nums.size-1){
            if(nums[i+1] < nums[i]) {
                if(modified) return false
                if(i == 0 || nums[i+1] >= nums[i-1]) nums[i] = nums[i+1]
                else nums[i+1] = nums[i]
                modified = true
            }
        }
        return true
    }
}
